File: debt tracker/Services/SampleDataService.swift
import Foundation
import SwiftData

enum SampleDataService {
    static func seedCategories(context: ModelContext) {
        for type in DebtCategoryType.allCases {
            let category = DebtCategory(from: type)
            context.insert(category)
        }
    }

    static func seedSampleData(context: ModelContext) {
        // Categories
        let categories = DebtCategoryType.allCases.map { DebtCategory(from: $0) }
        categories.forEach { context.insert($0) }

        // People
        let people = [
            Person(name: "Alex Johnson", phone: "555-0101"),
            Person(name: "Maria Garcia", email: "maria@email.com"),
            Person(name: "James Wilson"),
            Person(name: "Sarah Chen", phone: "555-0204", email: "sarah@email.com"),
            Person(name: "David Kim"),
            Person(name: "Emma Thompson", phone: "555-0306"),
        ]
        people.forEach { context.insert($0) }

        // Debts with varied states
        let calendar = Calendar.current
        let now = Date()

        let debts: [(String, Decimal, DebtDirection, Int, Int?, String?, DebtCategoryType)] = [
            ("Dinner at Nobu", 120.50, .owedToMe, 0, 14, nil, .food),
            ("Concert tickets", 85.00, .owedToMe, 1, 7, "Two tickets for the show", .personal),
            ("Rent share - Feb", 750.00, .iOwe, 2, -3, nil, .rent),
            ("Business supplies", 340.00, .owedToMe, 3, 30, "Office equipment", .business),
            ("Flight to NYC", 450.00, .iOwe, 4, 21, "Spring break trip", .travel),
            ("Textbooks", 200.00, .owedToMe, 1, nil, "Semester books", .education),
            ("Family dinner", 95.00, .iOwe, 5, -1, nil, .family),
            ("Gym membership", 60.00, .owedToMe, 0, 10, nil, .personal),
            ("Prescription meds", 180.00, .iOwe, 3, 5, nil, .medical),
            ("Grocery run", 67.30, .owedToMe, 2, nil, "Weekly groceries", .food),
            ("Uber rides", 42.00, .iOwe, 4, nil, nil, .travel),
            ("Birthday gift", 55.00, .owedToMe, 5, 3, "Group gift contribution", .personal),
        ]

        for (title, amount, direction, personIdx, dueDays, notes, catType) in debts {
            let dueDate = dueDays.map { calendar.date(byAdding: .day, value: $0, to: now) ?? now }
            let category = categories.first { $0.categoryType == catType }

            let debt = Debt(
                title: title,
                totalAmount: amount,
                direction: direction,
                person: people[personIdx],
                category: category,
                dueDate: dueDate,
                notes: notes
            )
            context.insert(debt)

            // Add some payments to certain debts
            if title == "Rent share - Feb" {
                let p1 = Payment(amount: 375.00, date: calendar.date(byAdding: .day, value: -5, to: now) ?? now, notes: "First half", debt: debt)
                context.insert(p1)
            } else if title == "Concert tickets" {
                let p1 = Payment(amount: 40.00, date: calendar.date(byAdding: .day, value: -2, to: now) ?? now, debt: debt)
                context.insert(p1)
            } else if title == "Grocery run" {
                let p1 = Payment(amount: 67.30, date: calendar.date(byAdding: .day, value: -1, to: now) ?? now, notes: "Paid in full", debt: debt)
                context.insert(p1)
            } else if title == "Family dinner" {
                let p1 = Payment(amount: 50.00, date: calendar.date(byAdding: .day, value: -3, to: now) ?? now, debt: debt)
                let p2 = Payment(amount: 45.00, date: now, notes: "Remaining balance", debt: debt)
                context.insert(p1)
                context.insert(p2)
            }
        }
    }

    static func clearAll(context: ModelContext) {
        do {
            try context.delete(model: Payment.self)
            try context.delete(model: Debt.self)
            try context.delete(model: Person.self)
            try context.delete(model: DebtCategory.self)
        } catch {
            print("Failed to clear data: \(error)")
        }
    }
}

File: android/app/src/main/java/com/kevingamez/debttracker/services/SampleDataService.kt
package com.kevingamez.debttracker.services

import com.kevingamez.debttracker.data.db.CategoryEntity
import com.kevingamez.debttracker.data.db.DebtEntity
import com.kevingamez.debttracker.data.db.PaymentEntity
import com.kevingamez.debttracker.data.db.PersonEntity
import com.kevingamez.debttracker.data.repository.DebtRepository
import com.kevingamez.debttracker.domain.model.DebtCategoryType
import com.kevingamez.debttracker.domain.model.DebtDirection
import java.math.BigDecimal
import java.time.Instant
import java.time.temporal.ChronoUnit
import javax.inject.Inject
import javax.inject.Singleton

/// Direct port of iOS [SampleDataService]. Same 12 fixtures, same payments,
/// so screenshots between platforms line up.
@Singleton
class SampleDataService @Inject constructor(private val repo: DebtRepository) {

    suspend fun seed() {
        // Categories — one per type, ensure-once.
        val categoryByType = DebtCategoryType.values().associateWith { type ->
            CategoryEntity(type = type).also { repo.ensureCategory(it) }
        }

        // People.
        val alex = PersonEntity(name = "Alex Johnson", phone = "555-0101")
        val maria = PersonEntity(name = "Maria Garcia", email = "maria@email.com")
        val james = PersonEntity(name = "James Wilson")
        val sarah = PersonEntity(name = "Sarah Chen", phone = "555-0204", email = "sarah@email.com")
        val david = PersonEntity(name = "David Kim")
        val emma = PersonEntity(name = "Emma Thompson", phone = "555-0306")
        listOf(alex, maria, james, sarah, david, emma).forEach { repo.upsertPerson(it) }

        val now = Instant.now()
        fun days(offset: Long): Instant = now.plus(offset, ChronoUnit.DAYS)

        data class Seed(
            val title: String,
            val amount: String,
            val direction: DebtDirection,
            val person: PersonEntity,
            val dueOffsetDays: Long?,
            val notes: String? = null,
            val category: DebtCategoryType,
        )

        val seeds = listOf(
            Seed("Dinner at Nobu", "120.50", DebtDirection.OWED_TO_ME, alex, 14, null, DebtCategoryType.FOOD),
            Seed("Concert tickets", "85.00", DebtDirection.OWED_TO_ME, maria, 7, "Two tickets for the show", DebtCategoryType.PERSONAL),
            Seed("Rent share - Feb", "750.00", DebtDirection.I_OWE, james, -3, null, DebtCategoryType.RENT),
            Seed("Business supplies", "340.00", DebtDirection.OWED_TO_ME, sarah, 30, "Office equipment", DebtCategoryType.BUSINESS),
            Seed("Flight to NYC", "450.00", DebtDirection.I_OWE, david, 21, "Spring break trip", DebtCategoryType.TRAVEL),
            Seed("Textbooks", "200.00", DebtDirection.OWED_TO_ME, maria, null, "Semester books", DebtCategoryType.EDUCATION),
            Seed("Family dinner", "95.00", DebtDirection.I_OWE, emma, -1, null, DebtCategoryType.FAMILY),
            Seed("Gym membership", "60.00", DebtDirection.OWED_TO_ME, alex, 10, null, DebtCategoryType.PERSONAL),
            Seed("Prescription meds", "180.00", DebtDirection.I_OWE, sarah, 5, null, DebtCategoryType.MEDICAL),
            Seed("Grocery run", "67.30", DebtDirection.OWED_TO_ME, james, null, "Weekly groceries", DebtCategoryType.FOOD),
            Seed("Uber rides", "42.00", DebtDirection.I_OWE, david, null, null, DebtCategoryType.TRAVEL),
            Seed("Birthday gift", "55.00", DebtDirection.OWED_TO_ME, emma, 3, "Group gift contribution", DebtCategoryType.PERSONAL),
        )

        for (s in seeds) {
            val debt = DebtEntity(
                title = s.title,
                totalAmount = BigDecimal(s.amount),
                direction = s.direction,
                personId = s.person.id,
                categoryId = categoryByType[s.category]?.id,
                dueDate = s.dueOffsetDays?.let { days(it) },
                notes = s.notes,
            )
            repo.upsertDebt(debt)

            // Same payment fixtures as iOS so the Dashboard tile totals match.
            when (s.title) {
                "Rent share - Feb" ->
                    repo.upsertPayment(PaymentEntity(debtId = debt.id, amount = BigDecimal("375.00"), date = days(-5), notes = "First half"))
                "Concert tickets" ->
                    repo.upsertPayment(PaymentEntity(debtId = debt.id, amount = BigDecimal("40.00"), date = days(-2)))
                "Grocery run" ->
                    repo.upsertPayment(PaymentEntity(debtId = debt.id, amount = BigDecimal("67.30"), date = days(-1), notes = "Paid in full"))
                "Family dinner" -> {
                    repo.upsertPayment(PaymentEntity(debtId = debt.id, amount = BigDecimal("50.00"), date = days(-3)))
                    repo.upsertPayment(PaymentEntity(debtId = debt.id, amount = BigDecimal("45.00"), date = now, notes = "Remaining balance"))
                }
            }
        }
    }
}
